redbot/cogs/cleanup/cleanup.py

```diff
@@ -152,7 +155,7 @@ def check(m):
         to_delete.append(ctx.message)
 
         reason = "{}({}) deleted {} messages containing '{}' in channel {}.".format(
-            author.name, author.id, len(to_delete), text, channel.id
+            author.name, author.id, humanize_number(len(to_delete)), text, channel.id
```

Comment:
This one should probably use `babel.format_decimal` directly with en_US as the locale, since the message will always be in English, we probably shouldn't put a number formatted for a different locale in there. Especially when we have context-specific locale :)

Reply:
I've added the `override_locale` arg to `humanize_number` this is to facilitate the overriding of the current bots permission while still being able to use `humanize_number` (So devs don't need to import babel and figure out it and to all the error handling we done here)